gemini/gemini.go

```diff
@@ -0,0 +1,70 @@
+package gemini
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/google/generative-ai-go/genai"
+	"google.golang.org/api/option"
+)
+
+var (
+	ErrNoCandidates = errors.New("no candidates found")
+	ErrNoResponse   = errors.New("no response found")
+	ErrNoGemApiKey  = errors.New("missing API key in environment variables")
+)
+
+func getAPIKey() (string, error) {
+	key := os.Getenv("GEM_API_KEY")
+	if key == "" {
+		return "", ErrNoGemApiKey
+	}
+	return key, nil
+}
+
+func extractFirstCandidate(candidates []*genai.Candidate) (string, error) {
+	for _, candidate := range candidates {
```

Comment:
maybe 
if len(candidates) != 0 && content := candidates[0]; content != nil && len(content.Parts) != nil {
return fmt.Sprintf("%v", content.Parts[0]), nil
}

Reply:
this worsens the readability

Reply:
decidanse ps